File: Desktop/ailydian-ultra-pro/public/docs/sdks/python/examples/lydian_iq.py
"""LyDian IQ API examples"""

import os
from lydian import Lydian
from lydian.types import KnowledgeQuery

lydian = Lydian(api_key=os.environ.get("LYDIAN_API_KEY"))

# 1. Ingest signal
signal = lydian.lydian_iq.ingest_signal(
    signal_type="sensor",
    source="temperature-sensor-001",
    content={"temperature": 72.5, "unit": "fahrenheit", "location": "Building A, Floor 3"},
    metadata={"sensor_model": "TempPro-200", "battery_level": 87}
)
print(f"Signal ingested: {signal.id}")

# 2. Batch ingest signals
signals = lydian.lydian_iq.batch_ingest_signals([
    {
        "type": "sensor",
        "source": "humidity-sensor-001",
        "content": {"humidity": 45, "unit": "percent"}
    },
    {
        "type": "sensor",
        "source": "air-quality-sensor-001",
        "content": {"pm25": 12.3, "aqi": 48}
    }
])
print(f"Batch ingested {len(signals)} signals")

# 3. Create knowledge entity
entity = lydian.lydian_iq.create_entity(
    entity_type="Building",
    name="Smart Office Tower A",
    properties={
        "address": "123 Main St",
        "floors": 20,
        "occupancy": 850,
        "energy_efficiency_rating": "A+"
    },
    relationships=[
        {"type": "contains", "target_id": "floor-3", "properties": {"floor_number": 3}}
    ]
)
print(f"Entity created: {entity.id}")

# 4. Query knowledge graph
query_result = lydian.lydian_iq.query_knowledge(
    KnowledgeQuery(
        query="Find all buildings with temperature sensors showing anomalies",
        entity_types=["Building", "Sensor"],
        filters={"Sensor.type": "temperature"},
        limit=50
    )
)
print(f"Found {len(query_result.get('entities', []))} entities")
print(f"Found {len(query_result.get('relationships', []))} relationships")

# 5. Generate insights
insights = lydian.lydian_iq.generate_insights(
    query="Analyze building energy consumption patterns",
    context={"time_range": "7d", "buildings": [entity.id]}
)
print(f"Generated {len(insights)} insights")

for insight in insights:
    print(f"\n[{insight.type}] {insight.title}")
    print(f"Confidence: {insight.confidence}%")
    print(f"Description: {insight.description}")
    if insight.actionable and insight.actions:
        print("Recommended actions:")
        for action in insight.actions:
            print(f"  - {action}")

File: Desktop/ailydian-ultra-pro/public/docs/sdks/typescript/examples/lydian-iq.ts
import { Lydian } from '@lydian/sdk';

const lydian = new Lydian({
  apiKey: process.env.LYDIAN_API_KEY,
});

async function lydianIQExample() {
  // 1. Ingest single signal
  const signal = await lydian.lydianIQ.ingestSignal({
    type: 'sensor',
    source: 'temperature-sensor-001',
    content: {
      temperature: 72.5,
      unit: 'fahrenheit',
      location: 'Building A, Floor 3',
    },
    metadata: {
      sensorModel: 'TempPro-200',
      batteryLevel: 87,
    },
  });

  console.log('Signal ingested:', signal.id);

  // 2. Batch ingest signals
  const signals = await lydian.lydianIQ.batchIngestSignals([
    {
      type: 'sensor',
      source: 'humidity-sensor-001',
      content: { humidity: 45, unit: 'percent' },
    },
    {
      type: 'sensor',
      source: 'air-quality-sensor-001',
      content: { pm25: 12.3, aqi: 48 },
    },
  ]);

  console.log(`Batch ingested ${signals.length} signals`);

  // 3. Create knowledge entity
  const entity = await lydian.lydianIQ.createEntity({
    type: 'Building',
    name: 'Smart Office Tower A',
    properties: {
      address: '123 Main St',
      floors: 20,
      occupancy: 850,
      energyEfficiencyRating: 'A+',
    },
    relationships: [
      {
        type: 'contains',
        targetId: 'floor-3',
        properties: { floor_number: 3 },
      },
    ],
  });

  console.log('Entity created:', entity.id);

  // 4. Query knowledge graph
  const queryResult = await lydian.lydianIQ.queryKnowledge({
    query: 'Find all buildings with temperature sensors showing anomalies',
    entityTypes: ['Building', 'Sensor'],
    filters: {
      'Sensor.type': 'temperature',
    },
    limit: 50,
  });

  console.log(`Found ${queryResult.entities.length} entities`);
  console.log(`Found ${queryResult.relationships.length} relationships`);
  console.log('Insights:', queryResult.insights);

  // 5. Generate insights
  const insights = await lydian.lydianIQ.generateInsights(
    'Analyze building energy consumption patterns',
    {
      timeRange: '7d',
      buildings: [entity.id],
    }
  );

  console.log(`Generated ${insights.length} insights`);

  insights.forEach(insight => {
    console.log(`\n[${insight.type}] ${insight.title}`);
    console.log(`Confidence: ${insight.confidence}%`);
    console.log(`Description: ${insight.description}`);
    if (insight.actionable && insight.actions) {
      console.log('Recommended actions:');
      insight.actions.forEach(action => console.log(`  - ${action}`));
    }
  });

  // 6. Get insights with filters
  const filteredInsights = await lydian.lydianIQ.getInsights({
    type: 'anomaly',
    minConfidence: 80,
    limit: 10,
  });

  console.log(`\nFound ${filteredInsights.data.length} high-confidence anomalies`);
}

lydianIQExample().catch(console.error);
